lightning/api_clnsocket_test.go

```diff
@@ -240,6 +241,62 @@ func TestClnGetChanInfo(t *testing.T) {
 	}
 }
 
+func TestClnGetNodeInfoFull(t *testing.T) {
+	info := clnData(t, "cln_nodeinfo_info")
+	funds := clnData(t, "cln_nodeinfo_funds")
+	channels := clnData(t, "cln_nodeinfo_channels")
+
+	_, api, closer := clnCommon(t, func(c net.Conn) {
+		buf := make([]byte, 0)
+
+		for {
+			tmp := make([]byte, BUFSIZE)
+			n, err := c.Read(tmp)
+			if err != nil {
+				if err == io.EOF {
+					break
+				}
+				t.Fatalf("Could not read request body: %v", err)
+			}
+
+			buf = append(buf, tmp[:n]...)
+			s := string(buf)
+
+			id := IDExtractor{}
+			err = json.Unmarshal(buf, &id)
+			if err != nil {
+				continue
+			} else {
+				// Reset buf
+				buf = make([]byte, 0)
+			}
+
+			reply := ""
+			if strings.Contains(s, "getinfo") {
+				reply = fmt.Sprintf(string(info), id.ID)
```

Comment:
having logic in tests is increasing complexity. Now you need test for test 🧐
Cant we just compare predefined input with predefined output ?

Reply:
Unfortunately it needs to have some logic